src/main/java/net/minecraftforge/fml/DatagenModLoader.java

```diff
@@ -39,14 +40,19 @@ public static boolean isRunningDataGen() {
         return runningDataGen;
     }
 
-    public static void begin(final Set<String> mods, final Path path, final java.util.Collection<Path> inputs, Collection<Path> existingPacks, final boolean serverGenerators, final boolean clientGenerators, final boolean devToolGenerators, final boolean reportsGenerator, final boolean structureValidator, final boolean flat) {
+    @Deprecated //TODO: Remove in 1.17
+    public static void begin(final Set<String> mods, final Path path, final Collection<Path> inputs, Collection<Path> existingPacks, final boolean serverGenerators, final boolean clientGenerators, final boolean devToolGenerators, final boolean reportsGenerator, final boolean structureValidator, final boolean flat) {
+        begin(mods, path, inputs, existingPacks, Collections.emptySet(), serverGenerators, clientGenerators, devToolGenerators, reportsGenerator, structureValidator, flat);
+    }
+
+    public static void begin(final Set<String> mods, final Path path, final Collection<Path> inputs, Collection<Path> existingPacks, Set<String> existingMods, final boolean serverGenerators, final boolean clientGenerators, final boolean devToolGenerators, final boolean reportsGenerator, final boolean structureValidator, final boolean flat) {
```

Comment:
Middle of the argument list..
This is a Forge class yes, but why?

Reply:
Because argument wise it makes more sense going next to existing packs than at the end after a ton of random boolean values, and once we hit 1.17 and the deprecated old one can be removed without worrying that someone is calling it for some reason from somewhere else, it really won't matter as it won't be the "middle" it will just be the "normal" place for it.